src/compute/cast/binary_to.rs

```diff
@@ -118,3 +118,44 @@ pub(super) fn binary_to_dictionary_dyn<O: Offset, K: DictionaryKey>(
     let values = from.as_any().downcast_ref().unwrap();
     binary_to_dictionary::<O, K>(values).map(|x| Box::new(x) as Box<dyn Array>)
 }
+
+fn fixed_size_to_offsets<O: Offset>(values_len: usize, fixed_size: usize) -> Offsets<O> {
+    let offsets = (0..(values_len + 1))
+        .step_by(fixed_size)
+        .map(|v| O::from_usize(v).unwrap())
+        .collect();
+    // Safety
+    // * every element is `>= 0`
+    // * element at position `i` is >= than element at position `i-1`.
+    unsafe { Offsets::new_unchecked(offsets) }
+}
+
+/// Conversion of large-binary
+pub fn fixed_size_binary_to_large_binary(
```

Comment:
Maybe refactor this into a generic with `<O: Offset>` to cover both cases at once?

Reply:
Right, good one. That I didn't see this.